cmd/snap-confine/seccomp-support.c

```diff
@@ -109,12 +136,73 @@ static void validate_bpfpath_is_safe(const char *path)
 	}
 }
 
+static void sc_cleanup_sock_fprog(struct sock_fprog *prog)
+{
+	free(prog->filter);
+	prog->filter = NULL;
+}
+
+static void sc_must_read_filter_from_file(FILE *file, uint32_t len_bytes, char *what, struct sock_fprog *prog)
+{
+	prog->len = len_bytes / sizeof(struct sock_filter);
+	prog->filter = (struct sock_filter *)malloc(len_bytes);
+	if (prog->filter == NULL) {
+		die("cannot allocate %u bytes of memory for %s seccomp filter ", len_bytes, what);
+	}
+	size_t num_read = fread(prog->filter, 1, len_bytes, file);
+	if (ferror(file)) {
+		die("cannot read %s filter", what);
+	}
+	if (num_read != len_bytes) {
+		die("short read for filter %s %zu != %i", what, num_read, len_bytes);
+	}
+}
+
+static FILE* sc_must_read_header_from_file(const char *profile_path, struct sc_seccomp_file_header *hdr)
```

Comment:
personal opinion here, this function both reads and verifies the header, maybe rename as appropriate or split out read logic from verification logic

Reply:
Excellent point, I pondered a bit and decided to not split the function and just renamed it. While I think in general split is better here we never want to open the file without validating so to reduce the risk of misuse I went this way. But happy to reconsider if others feel more strongly about this :)